data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/s3/S3SinkConfig.java

```diff
@@ -36,10 +36,21 @@ public class S3SinkConfig {
     private AwsAuthenticationOptions awsAuthenticationOptions;
 
     @JsonProperty("bucket")
-    @NotEmpty
     @Size(min = 3, max = 500, message = "bucket length should be at least 3 characters")
     private String bucketName;
 
+    @JsonProperty("bucket_selector")
```

Comment:
Please add `@JsonPropertyDescriptions` to these if possible. Can be a follow up PR

Reply:
Will do follow up PR